Source/ARTRealtime.m

```diff
@@ -180,12 +180,14 @@ @interface ARTRealtimeInternal ()
 NS_ASSUME_NONNULL_END
 
 const NSTimeInterval _immediateReconnectionDelay = 0.1;
+const NSTimeInterval _reachabilityReconnectionAttemptThreshold = 0.1;
 
 @implementation ARTRealtimeInternal {
     BOOL _resuming;
     BOOL _renewingToken;
     BOOL _shouldImmediatelyReconnect;
     ARTEventEmitter<ARTEvent *, ARTErrorInfo *> *_pingEventEmitter;
+    NSDate *_connectStartedAt;
```

Comment:
From what you're saying, we need a piece of state that we can use upon receiving a notification that the host is reachable, to determine how long it's been since we told the reachability class to start emitting reachability notifications. This concept doesn't seem to be directly related to connection state or the state transition, so I'm wondering if this name is the right one and whether you're setting up and tearing it down in the best place?

Reply:
https://github.com/ably/ably-cocoa/pull/1703/commits/08ba4c8fded2d94ec7efcecba95ccf979b9c80ca